src/main/java/com/zendesk/maxwell/MaxwellReplicator.java

```diff
@@ -202,6 +206,8 @@ private RowMapBuffer getTransactionRows() throws Exception {
 				continue;
 			}
 
+			setReplicatorPosition((AbstractRowEvent) v4Event);
```

Comment:
i think so.  as soon as we grab an event from the queue we're guaranteed to process it, so if replication stalls we should always restart after it, whether we restart here or in `getRows`